src/backend/parser/cypher_gram.y

```diff
@@ -118,7 +118,7 @@
 
 /* DELETE clause */
 %type <node> delete
-%type <boolean> detach_opt
+%type <boolean> optional_opt detach_opt
```

Comment:
The optional_opt should be defined with the rest of the match clause items starting on line 111,

Reply:
Moved on MATCH clause.